### PR TITLE
MEN-3730 OpenSSL via spacemonkey.

### PR DESCRIPTION
**Pull request by merlin-northern**
ChangeLog:none
Signed-off-by: Peter Grzybowski <peter@northern.tech>

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -280,6 +282,14 @@ func newHttpClient() *http.Client {
 	return &http.Client{}
 }
 
+func dialOpenSSL(ctx context.Context, network, addr string) (net.Conn, error) {
+	//contextSSL := ctx.Value("ssl").(*openssl.Ctx)
```
Comment (kacf) [inline client/client.go:286]: Remove?

**Comment by merlin-northern** [comment]
I had to move to https://github.com/mendersoftware/mender/pull/579 will tidy up

File: client/client.go [hunk 2]
```diff
@@ -295,6 +320,33 @@ func newHttpsClient(conf Config) (*http.Client, error) {
...
+			}
+
+			conn, err := openssl.Dial("tcp", addr, contextSSL, flags)
+			v := conn.VerifyResult()
+			if conn == nil || err != nil {
+				return nil, err
+			}
+			if v != openssl.Ok {
+				if v == openssl.CertHasExpired || v == openssl.DepthZeroSelfSignedCert {
+					return nil, errors.New(fmt.Sprintf("certificate has expired, openssl says: %d", v))
```
Comment (alfrunes) [inline client/client.go:344]: `errors.Errorf` or `fmt.Errorf`?

Comment (merlin-northern) [reply client/client.go:344]: sure Alf, thanks. (it will go into the final PR)

File: client/client.go (hunk 1 above)
Comment (merlin-northern) [reply client/client.go]: sure. it will be clean in the final PR.

**Comment by merlin-northern** [comment]
> This is the entire patch? Impressive! So once we knew what to do, it wasn't so difficult after all. :-)
> 
> Changing to OpenSSL is an important change, so this needs a good changelog. I would even consider a dedicated one, like `changelog: Switch to OpenSSL for all server communication.`

will do!

**Comment by merlin-northern** [comment]
superseded by https://github.com/mendersoftware/mender/pull/581

**Pull request closed without merging**